### DISCUSSION TITLE
How to find out whether field type is nullable?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/roslyn#66347] [Q&A] [answered]
I have this very simple snippet but so far I have not found out how to do check the syntax tree that the int has "?" assigned:

```
/// <summary>Value, or <c>null> otherwise.</summary>
private int? myInt;
```

I thought it was simply: `fieldDeclarationSyntax.Declaration.Type.IsNotNull` but it's probably something else. 

`IsNotNull` is not documented so I'm not sure what it is really.

Thank you

**Comment by CyrusNajmabadi** [comment, marked as answer]
check if the `.Type` is a `NullableTypeSyntax`.

**Reply by CyrusNajmabadi** [reply]
Note: this will only work for the `?` form, it will not work for `privaet Nullable<int> myInt;`.  If you want to work for both, you'll need symbols, not syntax.

**Reply by MartyIX** [reply]
Cool. Thank you!
